src/Symfony/Component/Workflow/TransitionBlockerList.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+/**
+ * A list of transition blockers.
+ */
+class TransitionBlockerList implements \IteratorAggregate, \Countable, \ArrayAccess
+{
+    /**
+     * @var TransitionBlocker[]
+     */
+    private $blockers = array();
+
+    /**
+     * Creates a new transition blocker list.
+     *
+     * @param TransitionBlocker[] $blockers The transition blockers to add to the list
+     */
+    public function __construct(array $blockers = array())
```

Comment:
You never use the array feature here. You always have to write `new TransitionBlockerList([$b]);`
You can change this to:

```php
public function __construct(TransitionBlocker $blocker = null)
{
    if ($blocker) {
        $this->add($blocker);
    }
}
```

Reply:
This constructor expects an array, because `TransitionBlockerList` is an array of `TransitionBlocker`s, but with a dedicated `::findByCode()` method. This method is literally the only reason why I created this class, because otherwise I would just be passing around `TransitionBlocker[]` (i.e. native arrays).

This is also, why there's `ConstraintViolation` and `ConstraintViolationList` in the `Validator` component.

I would be very surprised, if I discovered, that an array-like class can't be constructed with a native array (principle of least surprise).

Also, by accepting an array, I can construct this class with a collection of `TransitionBlocker`s or just a single `TransitionBlocker`. By _not_ accepting an array, I would need to construct this class with a `null` and then manually run the `TransitionBlockerList::add()` in a foreach loop in my own code, if I wanted to construct the class with a collection of `TransitionBlocker`s. This would be extremely annoying (poor DX).

I know, that in the Workflow code I never construct this list with more than 1 item. But taking into account the above, may I leave this constructor with an array argument, please? Having to wrap a single `TransitionBlocker` in an array, in order to construct this class, is really not an issue.